samples/subsys/smf/smf_calculator/src/main.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2024 Glenn Andrews
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "main.h"
+#include <zephyr/kernel.h>
+#include "smf_calculator_thread.h"
+#include <zephyr/drivers/display.h>
+#include <lvgl.h>
+#include <stdio.h>
+#include <string.h>
+#include <lvgl_input_device.h>
+#include <zephyr/logging/log.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+LOG_MODULE_REGISTER(main_app);
+
+K_MSGQ_DEFINE(output_msgq, CALCULATOR_STRING_LENGTH, 2, 1);
+
+#define CALCULATOR_BUTTON_LABEL_LENGTH 4
+struct calculator_button {
+	char label[CALCULATOR_BUTTON_LABEL_LENGTH];
+	struct calculator_event event;
+};
+
+/* clang-format off */
+struct calculator_button buttons[] = {
```

Comment:
To use a physical keypad? It can already be done using https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/input/zephyr%2Clvgl-button-input.yaml, it needs the x/y coordinate of the buttons though which means it depends on the specific screen used. Or this could use a dedicated input callback as an alternative to trigger `post_calculator_event`, that'd be pretty simple, don't think there's any upstream board with a keypad though, they are kind of out of fashion. :-)